lib/internal/test_runner/test.js

```diff
@@ -195,6 +209,18 @@ class Test extends AsyncResource {
       this.timeout = timeout;
     }
 
+    if (testNamePatterns !== null) {
+      // eslint-disable-next-line no-use-before-define
+      const match = this instanceof TestHook || ArrayPrototypeSome(
```

Comment:
can also test for `parent !== null`

Reply:
I'm not sure that works. `parent !== null` would allow all tests that don't match the pattern to run. I made the change locally and the test from this PR failed.